### PR TITLE
ECO-3544: use CSE encryption instead of CSF.

### PR DESCRIPTION
**Pull request by denis-gnusov**
- Developer(s): @denis-gnusov

- Ticket: https://spryker.atlassian.net/browse/ECO-3544

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Adyen                 | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Introduced `AdyenClient::getPaymentMethods()`.
- Introduced `ADYEN:SDK_CHECKOUT_SHOPPER_JS_URL` environment config to set URL to Adyen Checkout shopper SDK JS.
- Introduced `ADYEN:SDK_CHECKOUT_SHOPPER_CSS_URL` environment config to set URL to Adyen Checkout shopper SDK CSS.
- Introduced `ADYEN:SDK_CHECKOUT_SHOPPER_JS_INTEGRITY_HASH` to set Subresource Integrity (SRI) hash for Checkout shopper SDK JS.
- Introduced `ADYEN:SDK_CHECKOUT_SHOPPER_CSS_INTEGRITY_HASH` to set Subresource Integrity (SRI) hash for Checkout shopper SDK CSS.
- Introduced `ADYEN:SDK_ENVIRONMENT` to set environment name for SDK.
- Introduced `AdyenFacade::getPaymentMethods()`.
- Deprecated `AdyenConstants::SDK_CHECKOUT_SECURED_FIELDS_URL`.
- Disabled tslinter for deprecated `credit-card` mlecule.
- Adjusted `adyen-credit-card` molecule to provide properly working with Adyen Checkout shopper SDK:
   - Removed Adyen Checkout Secure Fields (CSF) JS SDK.
   - Integrated Adyen Web Components JS SDK. 
   - Introduced new styles for inner form elements into Adyen iframes;
   - Deprecated the `scriptSrc` data property;
   - Deprecated the `configKey` data property;
   - Removed unused `client-config-key` attribute;
   - Introduced new data properties: `sdkCheckoutShopperJsUrl`, `sdkCheckoutShopperCssUrl`, `sdkCheckoutShopperJsIntegrityHash`, `sdkCheckoutShopperCssIntegrityHash`, `paymentMethods`, `locale`;
   - Introduced new attributes: `payment-methods-response`, `container-id`, `origin-key`, `locale`, `environment`; 
   - Adjusted layout of `adyen-credit-card` molecule to properly working with Adyen Checkout shopper SDK;
   - Removed unused getters methods: `stylesIframesConfig` and `configKey`;
   - Removed unused methods `initIframes` and `gethostedIframesConfig`;
   - Removed unused `stylesConfig` object;
   - Moved code form the `readyCallback` method into the new `init` method;
   - Adjusted `mapEvents` and `onScriptLoad` methods according Spryker forntend convention;
   - Introduced new getters methods: `originKey`, `paymentMethodsResponse`, `containerId`, `locale` and `environment`;
   - Introduced new `createCheckout` method tio properly working with with Adyen Checkout shopper SDK;

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* eol=lf
```
Comment (dereuromark) [inline .gitattributes:2]: This seems to be an old template
How did you generate this?

Comment (dereuromark) [reply .gitattributes]: Can you help me understand where this template comes from? So we can fix the old sources for it?
The other files might have similar issues btw.

Comment (denis-gnusov) [reply .gitattributes]: I took this file from a fresh module from the core. As I see, all the `.gitattributes` files are the same except some outdated ones. Unfortunately, for the first time, I took an outdated one.

Comment (dereuromark) [reply .gitattributes]: Please always use the tools to create those files for now, we should never copy from existing modules.

File: src/SprykerEco/Yves/Adyen/Form/DataProvider/CreditCardFormDataProvider.php [hunk 2]
```diff
@@ -59,9 +69,32 @@ public function getData(AbstractTransfer $quoteTransfer): QuoteTransfer
...
+            CreditCardSubForm::SDK_CHECKOUT_PAYMENT_METHODS => $paymentMethodsRawApiResponse,
         ];
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\AdyenApiResponseTransfer $adyenApiResponseTransfer
+     *
+     * @return string
+     */
+    protected function buildPaymentMethodsRawApiResponse(AdyenApiResponseTransfer $adyenApiResponseTransfer): string
```
Comment (aleksey-kotsuba) [inline src/SprykerEco/Yves/Adyen/Form/DataProvider/CreditCardFormDataProvider.php:90]: Method name does not describe what it does.
Accepted param is not necessary.

**Comment by aleksey-kotsuba** [comment]
FE review is required.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts [hunk 3]
```diff
@@ -1,74 +1,69 @@
-declare var csf: any;
+/* tslint:disable: no-any */
+declare var AdyenCheckout: any;
```
Comment (andrew-maslov) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts:2]: Propose use `const`

Comment (zamlely) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts:2]:
```suggestion
declare const AdyenCheckout: any;
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 4]
```diff
@@ -5,38 +5,40 @@
...
 
+{#
+    @deprecated property "scriptSrc" is deprecate: kept for BC reasons, will be removed in next major.
+    @deprecated property "configKey" is deprecate: kept for BC reasons, will be removed in next major.
+#}
 {% define data = {
     form: required,
     scriptSrc: data.form.adyenCreditCard.vars.sdkUrl,
     configKey: data.form.adyenCreditCard.vars.sdkOriginKey,
+    sdkCheckoutShopperJsUrl: data.form.adyenCreditCard.vars.sdkCheckoutShopperJsUrl,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:16]:
```suggestion
    jsUrl: data.form.adyenCreditCard.vars.sdkCheckoutShopperJsUrl,
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 5]
```diff
@@ -5,38 +5,40 @@
...
+    sdkCheckoutShopperCssUrl: data.form.adyenCreditCard.vars.sdkCheckoutShopperCssUrl,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:17]:
```suggestion
    cssUrl: data.form.adyenCreditCard.vars.sdkCheckoutShopperCssUrl,
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 6]
```diff
@@ -5,38 +5,40 @@
...
+    sdkCheckoutShopperJsIntegrityHash: data.form.adyenCreditCard.vars.sdkCheckoutShopperJsIntegrityHash,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:18]:
```suggestion
    jsIntegrityHash: data.form.adyenCreditCard.vars.sdkCheckoutShopperJsIntegrityHash,
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 7]
```diff
@@ -5,38 +5,40 @@
...
+    sdkCheckoutShopperCssIntegrityHash: data.form.adyenCreditCard.vars.sdkCheckoutShopperCssIntegrityHash,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:19]:
```suggestion
    cssIntegrityHash: data.form.adyenCreditCard.vars.sdkCheckoutShopperCssIntegrityHash,
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 8]
```diff
@@ -1,3 +1,5 @@
+/* tslint:disable */
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts:1]: Why you disabled the whole file?

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss [hunk 9]
```diff
@@ -8,6 +8,23 @@
             }
         }
 
+        .adyen-checkout__input {
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss:11]: How these classes are related to this molecule?

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts (hunk 8 above)
Comment (zamlely) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts:1]: It's a deprecated component.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss (hunk 9 above)
Comment (zamlely) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss:11]: These classes are the inner stuff of the Adyen iframes. The goal of the molecule is rendering these iframes. This version of SDK have no opportunity to customize iframe styles from `ts`. Instead Adyen provides external css file and advise to override it to additional customization.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 10]
```diff
@@ -5,38 +5,40 @@
...
 } %}
 
 {% block body %}
+    {{ form_widget(data.form.adyenCreditCard) }}
+
     {% include molecule('script-loader') with {
+        class: config.jsName ~ '__script-loader',
         attributes: {
-            src: data.scriptSrc,
+            src: data.sdkCheckoutShopperJsUrl,
```
Comment (zamlely) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:39]:
```suggestion
            src: data.jsUrl,
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 11]
```diff
@@ -5,38 +5,40 @@
...
-            <div class="form__field col col--sm-6">
-                <span class="{{ config.name }}__input input input--expand" data-hosted-id="hostedExpiryDateField" data-cse="encryptedExpiryDate"></span>
-            </div>
 
-            <div class="form__field col col--sm-6">
-                <span class="{{ config.name }}__input input input--expand" data-hosted-id="hostedSecurityCodeField" data-cse="encryptedSecurityCode"></span>
-            </div>
-        </div>
-    </div>
+    <link rel="stylesheet" href="{{ data.sdkCheckoutShopperCssUrl }}" integrity="{{ data.sdkCheckoutShopperCssIntegrityHash }}" crossorigin="anonymous">
```
Comment (zamlely) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig:43]:
```suggestion
    <link rel="stylesheet" href="{{ data.cssUrl }}" integrity="{{ data.cssIntegrityHash }}" crossorigin="anonymous">
```

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss (hunk 9 above)
Comment (tolerants) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss:11]: Suggest to add a comment to describe it.

**Pull request merged**